src/server/cray/cfs/api/controllers/components.py

```diff
@@ -261,12 +455,33 @@ def put_component(component_id):
         return connexion.problem(
             status=400, title="Error parsing the data provided.",
             detail=str(err))
+    data["id"] = component_id
     data = _set_auto_fields(data)
     return DB.put(component_id, data), 200
 
 
 @dbutils.redis_error_handler
-def patch_component(component_id):
+def patch_component_v2(component_id):
+    """Used by the PATCH /components/{component_id} API operation"""
+    LOGGER.debug("PATCH /components/id invoked patch_component")
+    if component_id not in DB:
+        return connexion.problem(
+            status=404, title="Component could not found.",
+            detail="Component {} could not be found".format(component_id))
+    try:
+        data = connexion.request.get_json()
+    except Exception as err:
+        return connexion.problem(
+            status=400, title="Error parsing the data provided.",
+            detail=str(err))
+    data = dbutils.convert_data_from_v2(data, V2Component)
+    data = _set_auto_fields(data)
+    response_data = DB.patch(component_id, data, _update_handler)
+    return convert_component_to_v2(response_data), 200
+
+
+@dbutils.redis_error_handler
+def patch_component_v3(component_id):
     """Used by the PATCH /components/{component_id} API operation"""
     LOGGER.debug("PATCH /components/id invoked patch_component")
     if component_id not in DB:
```

Comment:
The "could not found" typo is also a couple lines down, just beyond where github will let me make a suggestion